coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationSubscription.kt

```diff
@@ -9,29 +9,35 @@ import org.bouncycastle.math.ec.ECPoint
 import java.security.KeyFactory
 import java.security.PublicKey
 import java.security.Security
+import java.time.Instant
 import java.util.*
 import javax.persistence.*
 
 @Entity
-@Table(name = "notification_subscription")
+@Table(
+    name = "notification_subscription",
+    indexes = [Index(columnList = "user_id", name = "notification_subscription_user_id_uix", unique = true)])
 class NotificationSubscription(
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     val id: Long? = null,
 
     @OneToOne
-    @JoinColumn(name = "user_id")
+    @JoinColumn(name = "user_id", foreignKey = ForeignKey(name = "notification_subscription_user_id_fkey"), nullable = false)
     val user: User,
 
-    @Column
+    @Column(columnDefinition = "TEXT", nullable = false)
```

Comment:
Isn't nullable = False always the case, or does it not understand Kotlin variable? Also TEXT sisnce it is a STRING or does that also have to be explicitly said?

Reply:
Hibernate uses VARCHAR(255) by default for Strings, but using TEXT allows us to have variable length strings without any impact on storage size.

nullable = false sets the column definition to be NOT NULL, while the Kotlin non-nullable value only enforces this at runtime, so it would lead to an exception (I think 🤔) if one of the columns turned out to be null. If the columns are ONLY updated through Hibernate this would never happen, but if we do something manually through `psql` or other tools, then it might be the case.